Common/Base/OpenGL/BaseManager.cpp

```diff
@@ -152,37 +80,86 @@ namespace base
 
 		createGui();
 
-		createInput((size_t)hWnd);
+		createInput();
 
-		createPointerManager((size_t)hWnd);
+		createPointerManager();
 
-		createScene();
+		// this needs to be called before createScene() since some demos require
+		// screen size to properly position the widgets
+		_windowResized(width, height);
 
-		_windowResized();
+		createScene();
 
 		return true;
 	}
 
 	void BaseManager::run()
 	{
-		MSG msg;
-		while (true)
+		while (!mExit)
 		{
-			while (PeekMessage(&msg, NULL, 0, 0, PM_REMOVE))
+			while (SDL_PollEvent(&mEvent) != 0)
 			{
-				TranslateMessage(&msg);
-				DispatchMessage(&msg);
+				switch (mEvent.type)
+				{
+				// keyboard events
+				case SDL_KEYDOWN:
+					mKeyCode = mEvent.key.keysym.sym;
+					if (mKeyCode == SDLK_BACKSPACE || mKeyCode == SDLK_RETURN) 
```

Comment:
This check should be removed. We want keycode events for all keys, not just backspace and return.